src/transaction/TransactionController.ts

```diff
@@ -458,6 +458,8 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
     const currentChainId = network?.state?.provider?.chainId;
     const index = transactions.findIndex(({ id }) => transactionID === id);
     const transactionMeta = transactions[index];
+    /* istanbul ignore next */
+    const nonce = transactionMeta.transaction?.nonce;
```

Comment:
Is it possible for `transactionMeta.transaction` to be falsey? I don't think this `?` needs to be here:

```suggestion
    const nonce = transactionMeta.transaction.nonce;
```

Reply:
By default `transactionMeta.transaction` does not have a `nonce` property. so it can be `undefined` in cases where one is not specified. This is how it functions in app today.

here's an example of  `transactionMeta.transaction` from a transaction I just sent:

```
{"from": "0xc7d3bfdea106b446cf9f2db354d496e6dd8b2525", "gas": "0x5208", "gasPrice": "0x2883354c00", "to": "0xd8c06dc1b55012f114cee3dace83f1d52240a876", "value": "0x9184e72a000"}
```

Reply:
Right, but even in that case `transactionMeta.transaction` is an object. The `?` here only ensures we can attempt accessing the `.nonce` property even if the part of the path before the operator is invalid (e.g. if `transaction` doesn't exist, or is not an object).

Reply:
ah, right! sorry my mistake. Updated.